pkg/server/server.go

```diff
@@ -411,36 +375,30 @@ type DiscoverOpts struct {
 // RegisterTasksAndViews generates a mapping of slug to task and view configs and stores the mappings in the server
 // state. Task registration must occur after the local dev server has started because the task discoverer hits the
 // /v0/tasks/getMetadata endpoint.
-func (s *Server) RegisterTasksAndViews(ctx context.Context, opts DiscoverOpts) (dev_errors.RegistrationWarnings, error) {
-	mergedResources, err := resources.MergeRemoteResources(ctx, s.state, nil)
-	if err != nil {
-		return dev_errors.RegistrationWarnings{}, errors.Wrap(err, "merging local and remote resources")
-	}
-	warnings, err := ValidateTasks(ctx, mergedResources, opts.Tasks)
+func (s *Server) RegisterTasksAndViews(ctx context.Context, opts DiscoverOpts) (map[string]dev_errors.AppError, error) {
+	unsupportedApps, err := ValidateTasks(ctx, opts.Tasks)
 	if err != nil {
-		return dev_errors.RegistrationWarnings{}, errors.Wrap(err, "validating task")
+		return nil, errors.Wrap(err, "validating task")
 	}
 	if opts.OverwriteAll {
 		// clear existing tasks, task errors, and views
 		s.state.TaskConfigs.ReplaceItems(map[string]discover.TaskConfig{})
-		s.state.AppCondition.ReplaceItems(map[string]state.AppCondition{})
 		s.state.ViewConfigs.ReplaceItems(map[string]discover.ViewConfig{})
 	}
-	now := time.Now()
+	// Always throw out all AppConditions
```

Comment:
Adjusted wording to clarify it's a cache.

```suggestion
	// Always invalidate the AppConditions cache.
```